api/integrations/integrator.py

```diff
@@ -40,56 +50,81 @@ def get_public_posts(self):
         """
         Queries foreign server for /posts
         """
-        response = self.request(request.get, "posts")
-
+        response = self.request(request.get, self.build_url("posts"))
         if response.status_code == 200:
-            return response.json()["posts"]
+            return self.prepare_post_data(response)
         else:
-            print "Error calling %s/posts" % (self.host)
+            print "Error calling %s" % (self.build_url("posts"))
             return []
 
-    def get_author(self, author_id):
+    def get_author(self, origin):
         """
         Queries foreign server for /author/:id
         """
-        endpoint = "author/%s" % author_id
-        response = self.request(request.get, endpoint).json()
-        author_serializer = AuthorSerializer(response)
-        return author_serializer.data
+        response = self.request(request.get, origin, {})
+        print response
+        if response.status_code == 200:
+            return self.prepare_author_data(response)
+        else:
+            return None
 
-    def get_author_posts(self, author_id):
+    def get_author_posts(self, origin, requestor=None):
         """
         Queries foreign server for /author/:id/posts
         """
-        endpoint = "author/%s/posts" % author_id
-        response = self.request(request.get, endpoint).json()["posts"]
-        posts_serializer = PostSerializer(response)
-        return posts_serializer.data
+        headers = {"UUID": str(requestor)}
```

Comment:
Don't ask. Requirement for Tamara's teams calls. I'm not going to bother trying to rig up anything more elegant for now.